### PR TITLE
feat: enforce fixture-backed e2e setup

### PR DESCRIPTION
**Pull request by harshal015**
## Description
app.clickup.com
Brief description of the changes made in this PR.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Refactoring (no functional changes)

## Testing

- [ ] Tests added/updated for new functionality
- [ ] All existing tests pass
- [ ] Manual testing completed
- [ ] Edge cases tested

## Checklist

- [ ] Code follows the project's style guidelines
- [ ] Self-review of code completed
- [ ] Documentation updated (if applicable)
- [ ] No breaking changes (or breaking changes are documented)
- [ ] Changes are backward compatible
- [ ] Linting passes (`npm run lint`)
- [ ] Tests pass (`npm test`)
- [ ] Type checking passes (`npm run type-check`)

## Related Issues

Closes #(issue number)

## Screenshots (if applicable)

Add screenshots to help explain your changes.

## Additional Notes

Any additional information that reviewers should know.

### CODE REVIEW

**Review by rvab — COMMENTED** [review]
Will review after @OmkarJ13 approval

File: packages/core/src/rules/enforce-fixture-setup.js [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+);
+
+const RULE_NAME = 'enforce-fixture-setup';
+const FIXTURE_FILE_NAME = 'fixture.yml';
+const SPEC_FILE_PATTERN = /\.spec\.[cm]?[jt]sx?$/u;
+const FIXTURE_SETUP_ALLOW_PATTERN = /fixture-setup-allow:\s*\S/u;
+const REPORT_STATES = new Set(['draft', 'submitted', 'sent_back', 'approved', 'processing', 'paid']);
+const COMPLETE_EXPENSE_STATES = new Set(['complete']);
+
+const alwaysSupported = () => true;
```
Comment (OmkarJ13) [inline packages/core/src/rules/enforce-fixture-setup.js:16]: what is this alwaysSupported?

File: packages/core/src/rules/enforce-fixture-setup.js [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    return false;
+  }
+
+  if (value.type === 'Literal') {
+    return true;
+  }
+
+  if (value.type === 'Identifier') {
+    return value.name === 'undefined';
+  }
```
[the remark this answers is not in the document]
Comment (OmkarJ13) [reply packages/core/src/rules/enforce-fixture-setup.js:237]: pls check this

File: packages/core/src/rules/enforce-fixture-setup.js [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+    type: 'problem',
+    docs: {
+      description: 'Require fixture.yml for E2E specs and prevent straightforward fixture-backed runtime setup',
+    },
+    schema: [],
+    messages: {
+      missingFixture:
+        'E2E spec "{{specFile}}" must have a sibling {{fixtureFile}}. Fixtures are resolved from the spec directory.',
+      useFixture:
+        '{{serviceName}}.{{methodName}}() has fixture support at "{{fixturePath}}" and its result is unused. Move this prerequisite setup to fixture.yml, or add an adjacent "fixture-setup-allow: <reason>" comment when runtime creation is intentional.',
```
Comment (OmkarJ13) [inline packages/core/src/rules/enforce-fixture-setup.js:296]: In what cases it can be intentional?

File: packages/core/src/rules/enforce-fixture-setup.js [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+function hasLiteralArgumentValue(node, argumentIndex, supportedValues) {
+  const argument = node.arguments[argumentIndex];
+  return argument?.type === 'Literal' && supportedValues.has(argument.value);
+}
+
+const serviceDefinitions = {
+  ReportsService: {
+    modules: new Set(['reports-service']),
+    methods: {
+      bulkCreate: {
```
Comment (OmkarJ13) [inline packages/core/src/rules/enforce-fixture-setup.js:27]: This looks too tightly coupled to the code, if method name changes the eslint rule stops working
could you check if theres a better way to enforce this?

**Review by OmkarJ13 — CHANGES_REQUESTED** [review]
added some comments
